18-websockets/classworks/Anna_Volodkova/web/src/pages/Chat/Chat.jsx

```diff
@@ -0,0 +1,100 @@
+import React, {useState} from 'react';
+import './Chat.css';
+import Message from './components/Message';
+
+function Chat(props) {
+  const socket = props.socket;
+  const [value, setValue] = useState('');
+  const [userName, setUserName] = useState('');
+
+  const [isTyping, setIsTyping] = useState(false);
+  const [typingUser, setTypingUser] = useState('')
+
+  const [messages, setMessages] = useState([]);
+
+  socket.on('get-message', data => {
+    setMessages([...messages, data]);
+  });
+
+  if(isTyping){
+    socket.emit('send-isTyping', userName);
```

Comment:
Your application makes a lot of requests to the server.
Use `debounce` instead. It will help you to limit the number of function invocation per some period of time.